mypy/checkexpr.py

```diff
@@ -1543,8 +1544,11 @@ def visit_cast_expr(self, expr: CastExpr) -> Type:
         """Type check a cast expression."""
         source_type = self.accept(expr.expr, type_context=AnyType(), allow_none_return=True)
         target_type = expr.type
-        if self.chk.options.warn_redundant_casts and is_same_type(source_type, target_type):
+        options = self.chk.options
+        if options.warn_redundant_casts and is_same_type(source_type, target_type):
             self.msg.redundant_cast(target_type, expr)
+        if options.disallow_implicit_any_types and has_any_from_silent_import(target_type):
+            self.msg.implicit_any_from_silent_import("Target type of cast", target_type, expr)
```

Comment:
Ah, that was my misunderstanding.  I saw you dissecting `Callable` types [here](https://github.com/python/mypy/pull/3405/files#diff-90cb1d334445b0300466f7da165ae183R615) and assumed that was because `has_any_from_silent_import` didn't properly handle them.  Taking a closer look, you're just trying to give people a better error message.  That seems reasonable!